Explain case when static object causes leaks. (#96)
dart-lang/leak_tracker: doc/TROUBLESHOOT.md

([not available](https://github.com/flutter/flutter/issues/127331) for Flutter tests),
or, if it is a test, by clicking `Debug` near the test name in IDE.

## Complicated cases

### 1. Static or global object causes notGCed leaks

> If you see notGCed leaks, where retaining path starts with global or static variable,

Reviewer: ```suggestion
If you see notGCed leaks, where the retaining path starts with a global or static variable,
```

this means that some objects were disposed, but references to them were never released.

> In this example, as `disposedD` is not needed any more, it should stop being referenced, together with disposal.

Reviewer: ```suggestion
In this example, as `disposedD` is not needed anymore, it should stop being referenced when disposed.
```

> If `A` and `B` are still needed, `B` should assign null to variable thet references `disposedD`.
> Otherwize, reference to first non-needed object on the path (`staticX`, `A` or `B`) should be relesed.

Reviewer: ```suggestion
If `A` and `B` are still needed, `B` should assign null to the variable that references `disposedD`.
Otherwise, the reference to the first non-needed object on the path (`staticX`, `A` or `B`) should be released.
```
